database/database.go

```diff
@@ -61,6 +62,36 @@ type Config struct {
 	SSLMode        string `yaml:"ssl_mode"`
 }
 
+type BeforeQueryFunc func(ctx context.Context, query string, args ...interface{}) context.Context
+
+func (extDb *Database) SetBeforeQuery(f BeforeQueryFunc) {
+	extDb.before = append(extDb.before, f)
+}
+
+func (extDb *Database) ExecWithContext(ctx context.Context, query string, args ...interface{}) (sql.Result, error) {
+	for _, f := range extDb.before {
+		ctx = f(ctx, query, args)
```

Comment:
I'm not sure, that it is a mistake, but anyway

consider the following code:

```package main

import (
	"fmt"
)

type ftype func(args ...interface{})

func exec(f ftype, args ...interface{}) {
	f(args)
}

func exec2(f ftype, args ...interface{}) {
	f(args...)
}

func main() {
	f := func(args ...interface{}) {
		for _, arg := range args {
			fmt.Printf("%+v\n", arg)
		}
	}
	exec(f, "a", "b")
	exec2(f, "a", "b")
}
```
In `exec2`, unlike `exec` the arguments are passed via `...`, and as a result you have several arguments, not a slice.
I'm not sure, that slices don't work in your case, but, if you still want different arguments, not a slice, you should use `...`